File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:    
        res = 0
        l, r = 0, len(height) - 1

        while l < r:
            w = r - l
            if height[l] <= height[r]:
                h = height[l]
                l += 1
            else:
                h = height[r]
                r -= 1
            
            area = h * w
            res = max(res, area)

        return res

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxArea = 0;
        int l = 0;
        int r = height.length - 1;
        int h = 0;
        int w = 0;
        int area = 0;
        while (l < r){
            w = r - l;
            if (height[l] < height[r]){
                h = height[l];
                l++;
            }
            else{
                h = height[r];
                r--;
            }

            
            area = h * w;
            if (area > maxArea){
                maxArea = area;
            }
        }
        return maxArea;
    }
}